clustered/client/src/main/java/org/ehcache/clustered/client/config/ClusteringServiceConfiguration.java

```diff
@@ -54,31 +55,38 @@
    * @throws NullPointerException if {@code clusterUri} is {@code null}
    * @throws IllegalArgumentException if {@code clusterUri} is not URI valid for cluster operations
    */
-  public ClusteringServiceConfiguration(final URI clusterUri) {
-    validateClusterUri(clusterUri);
-    this.clusterUri = clusterUri;
-    this.autoCreate = false;
-    this.serverConfiguration = null;
-    this.readOperationTimeout = DEFAULT_READ_OPERATION_TIMEOUT;
+  public ClusteringServiceConfiguration(URI clusterUri) {
+    this(clusterUri, DEFAULT_OPERATION_TIMEOUT);
   }
 
   /**
    * Creates a {@code ClusteringServiceConfiguration} from the properties provided.
    *
    * @param clusterUri the non-{@code null} URI identifying the cluster server
    * @param readOperationTimeout the {@code TimeoutDuration} specifying the time limit for clustered cache
-   *                            read operations; if {@code null}, the default value is used
+   *                            read operations
+   *
+   * @throws NullPointerException if {@code clusterUri} is {@code null}
+   * @throws IllegalArgumentException if {@code clusterUri} is not URI valid for cluster operations
+   *
+   * @deprecated Use {@link #ClusteringServiceConfiguration(URI, Timeouts)} instead
+   */
+  @Deprecated
+  public ClusteringServiceConfiguration(URI clusterUri, Duration readOperationTimeout) {
+    this(clusterUri, Timeouts.builder().setReadOperationTimeout(readOperationTimeout).build());
+  }
+
+  /**
+   * Creates a {@code ClusteringServiceConfiguration} from the properties provided.
+   *
+   * @param clusterUri the non-{@code null} URI identifying the cluster server
+   * @param operationTimeouts the {@link Timeouts} specifying the time limit for clustered cache operations
    *
    * @throws NullPointerException if {@code clusterUri} is {@code null}
    * @throws IllegalArgumentException if {@code clusterUri} is not URI valid for cluster operations
    */
-  public ClusteringServiceConfiguration(final URI clusterUri, final TimeoutDuration readOperationTimeout) {
```

Comment:
This is API breakage. We should keep these versions around and add new ones. There is not even a naming conflict so that should be easy.

Reply:
Yes, we can. The thought (I think with @chrisdennis ) was that it's probably unused right now so keeping compatibility is not important. But if you think it is, we can.

Reply:
Removed as discussed